unumpy/tests/test_numpy.py

```diff
@@ -229,6 +229,7 @@ def replace_args_kwargs(method, backend, args, kwargs):
         (np.nditer, ([[1, 2, 3]],), {}),
         (np.ndenumerate, ([[1, 2], [3, 4]],), {}),
         (np.ndindex, (3, 2, 1), {}),
+        (np.lib.Arrayterator, ([[1, 2], [3, 4]], 2), {}),
```

Comment:
`Arrayterator` is failing because it doesn't dispatch its 1st argument which needs to be an array. Also, it seems that the returned type is not an instance of `collections.abc.Iterator`. What's the best way to test this?

Reply:
>Arrayterator is failing because it doesn't dispatch its 1st argument which needs to be an array.

You will need to create a new metaclass which inherits from `ClassOverrideMeta` and does dispatch on the array arguments.

>Also, it seems that the returned type is not an instance of collections.abc.Iterator. 

Are you sure? For me it is an instance of Iterator.

Reply:
```python
In [3]: a = onp.arange(3*4*5*6).reshape(3, 4, 5, 6)                             

In [4]: a_itor = onp.lib.Arrayterator(a, 2)                                     

In [5]: isinstance(a_itor, collections.abc.Iterator)                            
Out[5]: False
```

It should be True.

Reply:
It should be an instance of `collections.abc.Iterable`. 😄 There's a slight difference.

Reply:
Oh, I misread. It's `Iterable` but not, as you said, an `Iterator`. In fact, I'm kind of surprised the others are `Iterator`. I've never used them with `next` rather than a for loop.

Reply:
A `for` loop accepts both, just an `Iterator` is exhausted after one use, whereas an `Iterable` can spawn new `Iterator`s that can iterate over the collection multiple times.